main/commons/src/main/java/org/cryptomator/common/mountpoint/MountPointChooser.java

```diff
@@ -138,20 +147,39 @@ default void cleanup(Path mountPoint) {
 	int getPriority();
 
 	/**
-	 * Called by the {@link MountPointChooserModule} to determine the execution order
-	 * of the registered MPCs. <b>Implementations usually should not override this
-	 * method.</b> This default implementation sorts the MPCs in ascending order
-	 * of their {@link #getPriority() priority.}<br>
+	 * Called by the {@link Volume} to determine the execution order of the registered MPCs.
+	 * <b>Implementations usually may not override this method.</b> This default implementation
+	 * sorts the MPCs in ascending order of their {@link #getPriority() priority.}<br>
 	 * <br>
 	 * <b>Original description:</b>
 	 * <p>{@inheritDoc}
 	 *
 	 * @implNote This default implementation sorts the MPCs in ascending order
-	 * of their {@link #getPriority() priority.}
+	 * of their {@link #getPriority() priority.} It depends on a correct implementation
+	 * of {@link #hashCode()} and {@link #equals(Object)} for the compared objects.
+	 * For this implementation "the least required correct implementation" maps
+	 * the following relationship correctly:<br>
+	 * If the objects {@code a} and {@code b} are {@code equal} as determined by
+	 * {@code #equals(Object)} then the {@code hashCodes} as determined by {@code #hashCode()}
+	 * of {@code a} and {@code b} must be equal. If the objects are {@code not equal}
+	 * then the {@code hashCodes} must be different.<br>
+	 * The default implementations of those two methods are sufficient.
+	 * @see Object#hashCode()
+	 * @see Object#equals(Object)
 	 */
 	@Override
 	default int compareTo(MountPointChooser other) {
-		//Sort by priority (ascending order)
-		return Integer.compare(this.getPriority(), other.getPriority());
+		Validate.notNull(other, "Other must not be null!");
```

Comment:
Use Guava instead of Apache Commons.
```suggestion
		Preconditions.checkNotNull(other, "Other must not be null!");
```

Reply:
Resolved by 7781572b6de33b2099a2f80d373992b287165de2